spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/config/JdbcPollingChannelAdapterParser.java

```diff
@@ -59,24 +61,24 @@ protected BeanMetadataElement parseSource(Element element, ParserContext parserC
 		}
 		String query = IntegrationNamespaceUtils.getTextFromAttributeOrNestedElement(element, "query", parserContext);
 		if (!StringUtils.hasText(query)) {
-			throw new BeanCreationException("The query attrbitue is required");
+			throw new BeanCreationException("The query attribute is required");
 		}
-		String update = IntegrationNamespaceUtils.getTextFromAttributeOrNestedElement(element, "update", parserContext);
 		if (refToDataSourceSet) {
 			builder.addConstructorArgReference(dataSourceRef);
 		}
 		else {
 			builder.addConstructorArgReference(jdbcOperationsRef);
 		}
 		builder.addConstructorArgValue(query);
+
 		IntegrationNamespaceUtils.setReferenceIfAttributeDefined(builder, element, "row-mapper");
 		IntegrationNamespaceUtils.setReferenceIfAttributeDefined(builder, element, "update-sql-parameter-source-factory");
 		IntegrationNamespaceUtils.setReferenceIfAttributeDefined(builder, element, "select-sql-parameter-source");
 		IntegrationNamespaceUtils.setValueIfAttributeDefined(builder, element, "max-rows-per-poll");
-		if (update != null) {
-			builder.addPropertyValue("updateSql", update);
-		}
+		IntegrationNamespaceUtils.setValueIfAttributeDefined(builder, element, "max-rows");
```

Comment:
We should either disallow specifying both, or change the schema docs to explain that if both are supplied, `max-rows` wins.